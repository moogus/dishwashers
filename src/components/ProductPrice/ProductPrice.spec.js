import React from 'react';
import { render } from '@testing-library/react';

import useGetProduct from '../../hooks/product/get-product';

import { ProductPrice } from './index';

jest.mock('../../hooks/product/get-product');

describe('<ProductPrice/>', () => {
  const renderContainer = () => render(<ProductPrice />);

  describe('when <ProductPrice/> is rendered', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      useGetProduct.mockReturnValue({
        price: 'Â£10000.00', additionalServices: ['some amazing stuff'],
      });
    });

    test('component renders as expected', () => {
      const { container } = renderContainer();
      expect(container).toMatchSnapshot();
    });
  });
});
