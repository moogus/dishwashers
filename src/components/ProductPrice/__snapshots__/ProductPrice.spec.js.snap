// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPrice/> when <ProductPrice/> is rendered component renders as expected 1`] = `
<div>
  <div
    class="sc-gsTCUz dLzFAf row"
  >
    <div
      class="col"
    >
      <p
        class="sc-bdfBwQ cYJQoO"
      >
        Â£10000.00
      </p>
      <p
        class="sc-dlfnbm keIcqY"
      >
        Claim an extra 3 years guarantee via redemption
      </p>
    </div>
  </div>
</div>
`;
